app/models/trip.rb

```diff
@@ -136,6 +137,10 @@ def clone_as_back_trip
     new_trip
   end
 
+  def before_actual_time
+    self.departure_time.hour < Time.now.hour || (self.departure_time.hour == Time.now.hour && self.departure_time.min <= Time.now.min)
```

Comment:
We can compare hours + minutes directly with `Time.now.strftime('%H%M').to_i`. Might guard it if `departure_time` `blank?`.

Reply:
Should it `raise` an error if `departure_time` is `blank` ?